File: backend/src/main/java/space/davids_digital/sweetie/orm/entity/ChatAdminEntity.java
package space.davids_digital.sweetie.orm.entity;

import jakarta.persistence.*;

@Entity
@IdClass(ChatAdminEntityId.class)
@Table(name = "chat_admins")
public class ChatAdminEntity {
    @Id
    @Column(name = "peer_id")
    public long peerId;

    @Id
    @Column(name = "user_id")
    public long userId;
}

File: backend/src/main/java/space/davids_digital/sweetie/orm/entity/ChatAdminEntity.kt
package space.davids_digital.sweetie.orm.entity

import jakarta.persistence.*

@Entity
@IdClass(ChatAdminEntityId::class)
@Table(
    name = "chat_admins",
    indexes = [
        Index(
            name = "chat_admins_peer_id_idx",
            columnList = "peer_id"
        )
    ]
)
class ChatAdminEntity {
    @Id
    @Column(name = "peer_id")
    var peerId: Long = 0

    @Id
    @Column(name = "user_id")
    var userId: Long = 0
}
